aiida/orm/data/__init__.py

```diff
@@ -471,6 +471,10 @@ class List(Data, MutableSequence):
     """
     _LIST_KEY = 'list'
 
+    def __init__(self):
+        super(List, self).__init__()
+        self._set_list(list())
```

Comment:
There is no way to initialize the list via constructor parameters, right?
If there was, this would be a problem, since this code forces the list to always be empty at construction.
An alternative might be to `_set_list` before calling the superclass constructor

Reply:
Still I think that one should accept kwargs params and pass them to the super.__init__ (e.g. if reloading from a UUID). Actually, with this change, would load_node(some_uuid) work, where some_uuid is the uuid of a previously stored List?

Reply:
No that does not work. We can pass args and kwargs to the `Data` constructor like you suggest, but do you also want a way to do something like `l = L([1,2,3])`?

Reply:
No, as you say this is not supported, but as there is a _set_list, this should work (after passing the *args and **kwargs, and putting the logic below in some kind of `if` that avoids setting it to an empty list if it was already set): 
`L(_list=[1,2,3])`, similarly to `ParameterData(dict={})`.
Even better, if there is no real reason to keep _set_list private, one can rename it to `set_list` and then it would be `L(list=[1,2,3])`.